build/configs/rtl8730e/rtl8730e_download.sh

```diff
@@ -125,50 +128,27 @@ function board_download()
 function board_erase()
 {
 	cd ${IMG_TOOL_PATH}
-	./upload_image_tool_linux "erase" $1 1 $2 $3
-}
-
-function board_check()
-{
-	cd ${IMG_TOOL_PATH}
-	./upload_image_tool_linux "check" $1
-	flashsz=$?
-	echo "Flash size is ${flashsz} MB"
-
-	for partidx in ${!parts[@]}; do
-		if [[ "${parts[$partidx]}" == "reserved" ]];then
-			continue
-		fi
-
-		if [[ "${parts[$partidx]}" == "ftl" ]];then
-			continue
+	if [[ $TTYDEV == *"USB"* ]]; then
+		#echo "UART erase"
+		./upload_image_tool_linux "erase" $1 1 $2 $3
+	fi
+
+	if [[ $TTYDEV == *"ACM"* ]]; then
+		#echo "USB erase"
+		#echo "Save info to USB_erase_setting.txt"
+		echo "$2" >> USB_erase_setting.txt
+		echo "$3" >> USB_erase_setting.txt
+		if [ "$4" == "$LAST_PARTITION" ]; then
```

Comment:
Is there any help or description of `$2`, `$3`, `$4`?
I don't know what the last partition works.

Reply:
$2 - erase start address
$3 - erase size in KB
as board_erase() is invoked from common_download.sh in a for loop, the tool will collect start address and erase size of all partition one by one, and start erase in the last loop.(USB download/erase flow)

$4 - partition name of the current loop
in pre_download(), assign LAST_PARTITION to the partition name in the last loop, eg.userfs. If last loop found, start erase process.
* pls note that the last partition name should be unique, otherwise may cause mismatch issue.

Another solution is to split into 2 functions, passing start address and erase size of all partition in the for loop in common_download.sh, then send erase cmd to 8730e_download.sh to process erase. However, this need some modifications in common_download.sh, which may affect download and erase of other boards.